RecoMuon/TrackerSeedGenerator/plugins/TSGForOIDNN.cc

```diff
@@ -0,0 +1,974 @@
+                                       const Propagator& propagatorAlong,
+                                       edm::ESHandle<Chi2MeasurementEstimatorBase>& estimator,
+                                       double errorSF,
+                                       unsigned int& hitlessSeedsMade,
+                                       unsigned int& numSeedsMade,
+                                       std::vector<TrajectorySeed>& out) const {
+  // create hitless seeds
+  LogTrace("TSGForOIDNN") << "TSGForOIDNN::makeSeedsWithoutHits: Start hitless" << std::endl;
+  std::vector<GeometricSearchDet::DetWithState> dets;
+  layer.compatibleDetsV(tsos, propagatorAlong, *estimator, dets);
+  if (!dets.empty()) {
+    auto const& detOnLayer = dets.front().first;
+    auto const& tsosOnLayer = dets.front().second;
+    LogTrace("TSGForOIDNN") << "TSGForOIDNN::makeSeedsWithoutHits: tsosOnLayer " << tsosOnLayer << std::endl;
+    if (!tsosOnLayer.isValid()) {
+      edm::LogInfo(theCategory_) << "ERROR!: Hitless TSOS is not valid!";
+    } else {
+      dets.front().second.rescaleError(errorSF);
+      PTrajectoryStateOnDet const& ptsod =
+          trajectoryStateTransform::persistentState(tsosOnLayer, detOnLayer->geographicalId().rawId());
+      TrajectorySeed::RecHitContainer rHC;
+      out.push_back(TrajectorySeed(ptsod, rHC, oppositeToMomentum));
+      LogTrace("TSGForOIDNN") << "TSGForOIDNN::makeSeedsWithoutHits: TSOS (Hitless) done " << std::endl;
+      hitlessSeedsMade++;
+      numSeedsMade++;
+    }
+  }
+}
+
+//
+// Find hits on a given layer (TOB or TEC) and create seeds from updated TSOS with hit
+//
+void TSGForOIDNN::makeSeedsFromHits(const GeometricSearchDet& layer,
+                                    const TrajectoryStateOnSurface& tsos,
+                                    const Propagator& propagatorAlong,
+                                    edm::ESHandle<Chi2MeasurementEstimatorBase>& estimator,
+                                    edm::Handle<MeasurementTrackerEvent>& measurementTracker,
+                                    unsigned int& hitSeedsMade,
+                                    unsigned int& numSeedsMade,
+                                    unsigned int& layerCount,
+                                    std::vector<TrajectorySeed>& out) const {
+  if (layerCount > numOfLayersToTry_)
+    return;
+
+  const TrajectoryStateOnSurface& onLayer(tsos);
+
+  std::vector<GeometricSearchDet::DetWithState> dets;
+  layer.compatibleDetsV(onLayer, propagatorAlong, *estimator, dets);
+
+  // Find Measurements on each DetWithState
+  LogTrace("TSGForOIDNN") << "TSGForOIDNN::makeSeedsFromHits: Find measurements on each detWithState  " << dets.size()
+                          << std::endl;
+  std::vector<TrajectoryMeasurement> meas;
+  for (std::vector<GeometricSearchDet::DetWithState>::iterator it = dets.begin(); it != dets.end(); ++it) {
+    MeasurementDetWithData det = measurementTracker->idToDet(it->first->geographicalId());
```

Comment:
```suggestion
  for (auto const& detI : dets) {
    MeasurementDetWithData det = measurementTracker->idToDet(detI.first->geographicalId());
```
range loops are preferred

Reply:
Done!